internal/node/node_loader.js

```diff
@@ -186,7 +200,29 @@ function loadAsDirectorySync(res) {
 }
 
 function resolveManifestFile(res) {
-  return runfilesManifest[res] || runfilesManifest[res + '.js'];
+  const maybe = runfilesManifest[res] || runfilesManifest[res + '.js'];
+  if (maybe) {
+    return maybe;
+  }
+  // Look for tree artifacts that match and update
+  // the runfiles with files that are in the tree artifact.
```

Comment:
This doesn't sound right. If a file is present in the runfiles folder but not in the manifest, something is wrong.

Can you give me some context on why this needs to be done?

Reply:
Bazel allows you declare declare directories as outputs (aka tree artifacts). This applies to code-splitting as there is an unknown number of chunks that will be outputted when an app is code split so the entire directory is just declared as the output:

```
code_split_es6_output_dir = ctx.actions.declare_directory(ctx.label.name + ".cs.es6")
```

If this directory is listed as an input to a nodejs_binary/nodejs_test than only the directory name is listed in the RUNFILES_MANIFEST.

So when trying to resolve something within the directory such as "bundle.cs.es6/main.js" it can't be found without some extra leg work of determining that `bundle.cs.es6` is in the manifest and scanning what files are under that directory.

Reply:
Thank you for the explanation. I did not know about tree artifacts.

I was worried about this adding files to the runfiles that should not be there, but now I don't think that can happen since it only adds the descendents of a folder that is in the runfiles.